src/types/algorand-client.spec.ts

```diff
@@ -98,7 +99,7 @@ describe('AlgorandClient', () => {
       sender: alice.addr,
       appId: appId,
       method: appClient.appClient.getABIMethod('txnArg')!,
-      args: [{ type: 'pay' as const, sender: alice.addr, receiver: alice.addr, amount: algokit.microAlgos(0) }],
+      args: [algorand.transactions.payment({ sender: alice.addr, receiver: alice.addr, amount: algokit.microAlgos(0) })],
```

Comment:
By allowing `Promise<Transaction>` we get to have a similarly nice syntax, but without leaking the internal `Txn` type.